app/src/main/kotlin/com/hippo/ehviewer/ui/main/GalleryComment.kt

```diff
@@ -68,21 +65,46 @@ fun GalleryCommentCard(
                     },
                 )
             }
-            val textColor = LocalContentColor.current.toArgb()
-            val linkTextColor = MaterialTheme.colorScheme.primary.toArgb()
-            AndroidView(
-                factory = { context ->
-                    LinkifyTextView(context) { currentSpan = it }.apply {
-                        setTextColor(textColor)
-                        setLinkTextColor(linkTextColor)
-                    }
-                },
+            Column(
                 modifier = Modifier.constrainAs(commentRef) {
                     start.linkTo(parent.start)
                     top.linkTo(userRef.bottom, margin = 8.dp)
                 },
-                update = update,
-            )
+            ) {
+                ProvideTextStyle(value = MaterialTheme.typography.bodyMedium) {
+                    val linkColor = MaterialTheme.colorScheme.primary
+                    val processed = processComment(
+                        comment,
+                        TextLinkStyles(style = SpanStyle(color = linkColor)),
+                    ) { link ->
+                        check(link is LinkAnnotation.Url)
+                        onUrlClick(link.url)
+                    }
+                    val list = breakToTextAndUrl(comment.comment, processed)
+                    if (showImage) {
+                        list.forEach {
+                            when (it) {
+                                is Either.Left -> AsyncImage(
+                                    model = it.value,
+                                    contentDescription = null,
+                                )
+                                is Either.Right -> Text(text = it.value)
+                            }
+                        }
+                    } else {
+                        Text(
+                            text = list.fastFold(AnnotatedString("")) { a, e ->
```

Comment:
We don't need to break text if showImage = false.

Reply:
text may contains IMAGE_OBJ

Reply:
Done